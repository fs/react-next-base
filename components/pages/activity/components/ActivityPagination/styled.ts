import styled, { css } from 'styled-components';

import { component as LeftPointer } from 'public/images/icons/left-pointer.svg';
import { component as RightPointer } from 'public/images/icons/right-pointer.svg';

export const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 1rem;
`;

export const pointerIconStyles = css`
  width: 15px;
`;
export const LeftPointerIcon = styled(LeftPointer)`
  ${pointerIconStyles}
`;
export const RightPointerIcon = styled(RightPointer)`
  ${pointerIconStyles}
`;

export const prevButtonStyles = css`
  border-right-width: 0;
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
`;
export const nextButtonStyles = css`
  border-left-width: 0;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
`;

export const PageNumber = styled.span`
  padding: 0.25rem 0.7rem;
  border: 2px solid ${({ theme }) => theme.colors.grey};
  font-size: 1.1rem;
`;
