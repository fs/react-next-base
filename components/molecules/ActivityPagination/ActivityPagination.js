import React from 'react';
import styled, { css } from 'styled-components';

import LeftPointer from 'public/images/icons/left-pointer.svg';
import RightPointer from 'public/images/icons/right-pointer.svg';
import PaginationButton from 'components/atoms/PaginationButton';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 1rem;
`;

const pointerIconStyles = css`
  width: 15px;
`;
const LeftPointerIcon = styled(LeftPointer)`
  ${pointerIconStyles}
`;
const RightPointerIcon = styled(RightPointer)`
  ${pointerIconStyles}
`;

const prevButtonStyles = css`
  border-right-width: 0;
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
`;
const nextButtonStyles = css`
  border-left-width: 0;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
`;

const PageNumber = styled.span`
  padding: 0.25rem 0.7rem;
  border: 2px solid ${({ theme }) => theme.colors.grey};
  font-size: 1.1rem;
`;

const ActivityPagination = ({ pageInfo, setBeforeCursor, setAfterCursor, pageNumber, setPageNumber }) => {
  const { hasPreviousPage, hasNextPage, startCursor, endCursor } = pageInfo;

  const goToPrevPage = () => {
    setBeforeCursor(startCursor);
    setAfterCursor(undefined);
    setPageNumber(prevPageNumber => prevPageNumber - 1);
  };

  const goToNextPage = () => {
    setAfterCursor(endCursor);
    setBeforeCursor(undefined);
    setPageNumber(prevPageNumber => prevPageNumber + 1);
  };

  return (
    <Wrapper data-testid="activity-pagination">
      <PaginationButton disabled={!hasPreviousPage} onClick={() => goToPrevPage()} customStyles={prevButtonStyles}>
        <LeftPointerIcon />
      </PaginationButton>
      <PageNumber>{pageNumber}</PageNumber>
      <PaginationButton disabled={!hasNextPage} onClick={() => goToNextPage()} customStyles={nextButtonStyles}>
        <RightPointerIcon />
      </PaginationButton>
    </Wrapper>
  );
};

export default ActivityPagination;
