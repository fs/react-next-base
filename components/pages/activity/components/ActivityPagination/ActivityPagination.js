import React from 'react';

import PaginationButton from 'components/shared/atoms/PaginationButton';

import { Wrapper, LeftPointerIcon, PageNumber, RightPointerIcon, prevButtonStyles, nextButtonStyles } from './styled';

const ActivityPagination = ({ pageInfo, setBeforeCursor, setAfterCursor, pageNumber, setPageNumber }) => {
  const { hasPreviousPage, hasNextPage, startCursor, endCursor } = pageInfo;

  const goToPrevPage = () => {
    setBeforeCursor(startCursor);
    setAfterCursor(undefined);
    setPageNumber((prevPageNumber) => prevPageNumber - 1);
  };

  const goToNextPage = () => {
    setAfterCursor(endCursor);
    setBeforeCursor(undefined);
    setPageNumber((prevPageNumber) => prevPageNumber + 1);
  };

  return (
    <Wrapper data-testid="activity-pagination">
      <PaginationButton disabled={!hasPreviousPage} onClick={() => goToPrevPage()} customStyles={prevButtonStyles}>
        <LeftPointerIcon />
      </PaginationButton>
      <PageNumber>{pageNumber}</PageNumber>
      <PaginationButton
        testId="next-pagination"
        disabled={!hasNextPage}
        onClick={() => goToNextPage()}
        customStyles={nextButtonStyles}
      >
        <RightPointerIcon />
      </PaginationButton>
    </Wrapper>
  );
};

export default ActivityPagination;
