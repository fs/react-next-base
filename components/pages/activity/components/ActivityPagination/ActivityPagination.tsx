import React from 'react';

import PaginationButton from 'components/shared/atoms/PaginationButton';

import { PageInfo } from 'interfaces/activityType';

import { Wrapper, LeftPointerIcon, PageNumber, RightPointerIcon, prevButtonStyles, nextButtonStyles } from './styled';

type ActivityPaginationType = {
  pageInfo: PageInfo;
  setBeforeCursor: React.Dispatch<React.SetStateAction<string | undefined>>;
  setAfterCursor: React.Dispatch<React.SetStateAction<string | undefined>>;
  setPageNumber: React.Dispatch<React.SetStateAction<number>>;
  pageNumber: number;
};

const ActivityPagination = ({
  pageInfo,
  setBeforeCursor,
  setAfterCursor,
  pageNumber,
  setPageNumber,
}: ActivityPaginationType) => {
  const { hasPreviousPage, hasNextPage, startCursor, endCursor } = pageInfo;

  const goToPrevPage = () => {
    setBeforeCursor(startCursor);
    setAfterCursor(undefined);
    setPageNumber((prevPageNumber) => prevPageNumber - 1);
  };

  const goToNextPage = () => {
    setAfterCursor(endCursor);
    setBeforeCursor(undefined);
    setPageNumber((prevPageNumber) => prevPageNumber + 1);
  };

  return (
    <Wrapper data-testid="activity-pagination">
      <PaginationButton disabled={!hasPreviousPage} onClick={() => goToPrevPage()} customStyles={prevButtonStyles}>
        <LeftPointerIcon />
      </PaginationButton>

      <PageNumber>{pageNumber}</PageNumber>

      <PaginationButton
        testId="next-pagination"
        disabled={!hasNextPage}
        onClick={() => goToNextPage()}
        customStyles={nextButtonStyles}
      >
        <RightPointerIcon />
      </PaginationButton>
    </Wrapper>
  );
};

export default ActivityPagination;
