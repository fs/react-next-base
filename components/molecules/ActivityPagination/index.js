import ActivityPagination from './ActivityPagination';

export default ActivityPagination;
