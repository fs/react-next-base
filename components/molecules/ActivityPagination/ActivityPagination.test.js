import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

import renderWithTheme from '__tests__/helpers/renderWithTheme';

import ActivityPagination from 'components/molecules/ActivityPagination';

describe('ActivityPagination', () => {
  test('should render correctly', () => {
    // Arrange
    const mockTestId = 'test-activity-pagination';
    const mockPageInfo = 1;

    // Act
    render(
      renderWithTheme(
        <ActivityPagination
          pageInfo={mockPageInfo}
          pageNumber={pageNumber}
          setPageNumber={setPageNumber}
          testId={mockTestId}
        />,
      ),
    );

    // Assert
  });
});
