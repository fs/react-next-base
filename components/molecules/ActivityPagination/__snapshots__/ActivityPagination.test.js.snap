// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityPagination should render correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem;
  background-color: transparent;
  border: 2px solid #ccc;
  cursor: default;
  border-right-width: 0;
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
}

.c1 svg {
  fill: #ccc;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem;
  background-color: transparent;
  border: 2px solid #ccc;
  border-left-width: 0;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.c4 {
  background-color: transparent;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c4:hover {
  background-color: rgba(0,0,0,0.05);
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1rem;
}

.c2 {
  width: 15px;
}

.c5 {
  width: 15px;
}

.c3 {
  padding: 0.25rem 0.7rem;
  border: 2px solid #ccc;
  font-size: 1.1rem;
}

<div
  class="c0"
  data-testid="activity-pagination"
>
  <button
    class="c1"
    disabled=""
    type="button"
  >
    <svg
      class="c2"
      viewBox="0 0 492 492"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M198.608 246.104L382.664 62.04c5.068-5.056 7.856-11.816 7.856-19.024 0-7.212-2.788-13.968-7.856-19.032l-16.128-16.12C361.476 2.792 354.712 0 347.504 0s-13.964 2.792-19.028 7.864L109.328 227.008c-5.084 5.08-7.868 11.868-7.848 19.084-.02 7.248 2.76 14.028 7.848 19.112l218.944 218.932c5.064 5.072 11.82 7.864 19.032 7.864 7.208 0 13.964-2.792 19.032-7.864l16.124-16.12c10.492-10.492 10.492-27.572 0-38.06L198.608 246.104z"
      />
    </svg>
  </button>
  <span
    class="c3"
  >
    1
  </span>
  <button
    class="c4"
    data-cy="next-pagination"
    data-testid="next-pagination"
    type="button"
  >
    <svg
      class="c5"
      viewBox="0 0 492.004 492.004"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M382.678 226.804L163.73 7.86C158.666 2.792 151.906 0 144.698 0s-13.968 2.792-19.032 7.86l-16.124 16.12c-10.492 10.504-10.492 27.576 0 38.064L293.398 245.9l-184.06 184.06c-5.064 5.068-7.86 11.824-7.86 19.028 0 7.212 2.796 13.968 7.86 19.04l16.124 16.116c5.068 5.068 11.824 7.86 19.032 7.86s13.968-2.792 19.032-7.86L382.678 265c5.076-5.084 7.864-11.872 7.848-19.088.016-7.244-2.772-14.028-7.848-19.108z"
      />
    </svg>
  </button>
</div>
`;
